### PR TITLE
Disfavor current working directory when looking up executables

### PR DESCRIPTION
**Pull request by stevapple**
Calling the driver from the `/usr/bin` directory of a toolchain will use that toolchain even if the called driver is from another toolchain. I thought this should be regarded as a bug?

### CODE REVIEW

File: Sources/SwiftDriver/Toolchains/Toolchain.swift [hunk 1]
```diff
@@ -160,14 +160,14 @@ extension Toolchain {
...
+              let path = lookupExecutablePath(filename: executableName(executable), currentWorkingDirectory: nil, searchPaths: [toolDir]) {
       // Looking for tools from the tools directory.
       return path
-    } else if let path = lookupExecutablePath(filename: executableName(executable), searchPaths: [executableDir]) {
+    } else if let path = lookupExecutablePath(filename: executableName(executable), currentWorkingDirectory: nil, searchPaths: [executableDir]) {
       return path
     } else if let path = try? xcrunFind(executable: executableName(executable)) {
       return path
-    } else if !["swift-frontend", "swift", "swift-frontend.exe", "swift.exe"].contains(executable),
+    } else if !["swift", "swift-frontend"].map({$0 + executableFileSuffix}).contains(executable),
```
Comment (compnerd) [inline Sources/SwiftDriver/Toolchains/Toolchain.swift:170]: This breaks windows - the original has both the suffixed and unsuffixed variant.  I think that leaving it spelt out is better.

**Comment by stevapple** [comment]
Anyone to review?

**Comment by stevapple** [comment]
Ping again

**Comment by cltnschlosser** [comment]
@swift-ci please test

**Comment by cltnschlosser** [comment]
This seems reasonable to me, but idk if it breaks something I don’t know about

**Comment by stevapple** [comment]
Ping for someone to review & merge

**Comment by stevapple** [comment]
@DougGregor Sorry for pulling you in if you’re not in charge of this — could any of you and your folks review and merge this PR?

**Comment by stevapple** [comment]
Ping again

File: Sources/SwiftDriver/Toolchains/Toolchain.swift [hunk 2]
```diff
@@ -160,10 +160,10 @@ extension Toolchain {
     if let overrideString = envVar(forExecutable: executableName(executable)) {
       return try AbsolutePath(validating: overrideString)
     } else if let toolDir = toolDirectory,
-              let path = lookupExecutablePath(filename: executableName(executable), searchPaths: [toolDir]) {
+              let path = lookupExecutablePath(filename: executableName(executable), currentWorkingDirectory: nil, searchPaths: [toolDir]) {
       // Looking for tools from the tools directory.
       return path
-    } else if let path = lookupExecutablePath(filename: executableName(executable), searchPaths: [executableDir]) {
+    } else if let path = lookupExecutablePath(filename: executableName(executable), currentWorkingDirectory: nil, searchPaths: [executableDir]) {
```
Comment (artemcm) [inline Sources/SwiftDriver/Toolchains/Toolchain.swift:166]: As far as I can tell, this call will still result in us looking in the cwd based on `executableDir` anyway so this change looks like it ought to be safe. 

@stevapple could you please add a test for this behavior?
You can instantiate the driver with `compilerExecutableDir` pointing to a faux driver in a directory with a faux frontend and also pass in a tools dir, and make sure we prefer the tools dir frontend, for example.

**Review by artemcm — APPROVED** [review]

**Review by artemcm — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Please ensure that you run the test suite on Windows.  I think that this test might fail due to the executable names.

**Comment by artemcm** [comment]
cc @shahmishal

**Comment by stevapple** [comment]
> Please ensure that you run the test suite on Windows. I think that this test might fail due to the executable names.

Nice catch. Fixed.

**Comment by stevapple** [comment]
It’s somehow confusing because I cannot reproduce CI failure locally, so I may need someone’s help here to find out what happens.

**Comment by stevapple** [comment]
Anyone to run test again?

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by stevapple** [comment]
I cannot reproduce the failure both on Linux and macOS😢 Need help here to find out what happens on CI.

**Comment by stevapple** [comment]
I guess I found the reason👀 Anyone to test again & merge?

**Comment by artemcm** [comment]
@swift-ci please test

**Pull request merged**